src/wp-includes/media.php

```diff
@@ -4123,7 +4123,22 @@ function _wp_image_editor_choose( $args = array() ) {
 	 *                                'WP_Image_Editor_Imagick', 'WP_Image_Editor_GD'.
 	 */
 	$implementations = apply_filters( 'wp_image_editors', array( 'WP_Image_Editor_Imagick', 'WP_Image_Editor_GD' ) );
-	$supports_input  = false;
+
+	$editors = wp_cache_get( 'wp_image_editor_choose', 'image_editor' );
+
+	if ( ! is_array( $editors ) ) {
+		$editors = array();
+	}
+
+	// Cache the chosen editor implementation based on specific args and available implementations.
+	$cache_key = md5( serialize( array( $args, $implementations ) ) );
+
+	if ( isset( $editors[ $cache_key ] ) ) {
```

Comment:
I'm curious what's the rationale for nesting all the cache values in a single cache key "bucket". In most places in Core we store individual values per cache key, and the dynamic portion becomes part of the actual cache key.

For example the actual cache key would be:
```php
$cache_key = 'wp_image_editor_choose_' . md5( serialize( array( $args, $implementations ) ) );
```

Since these caches would expire after 1 day anyway, I don't think there's any harm in storing in individual cache keys. This approach here might work too, but there's potential concerns about scalability and diverging from how I believe we typically handle such caches in Core.

Reply:
I didn't have any real intention when I chose to implement it this way, but I can see a few benefits. 

Generally, including the hash as part of the name is done for the purpose of avoiding stale caches (i.e., if the values that are hashed change, the `wp_get_cache()` call will result in a cache miss). That's not really a concern for this cache because the need to store the data for long periods while avoiding returning stale values is not that high.

This specific function can get called multiple times during the same run with different args. Having the results stored in multiple distinct caches would mean that we need to load separate cache values for each specific set of args. By saving all of the values to one cache, all of the cached values are loaded to memory the first time `_wp_image_editor_choose()` is called, and can be accessed from memory on each subsequent call.

Another minor benefit is that it makes it a bit easier to work with a cache that has a static name in the DB instead of a dynamic one. As a theoretical example you could do something like `wp cache get wp_image_editor_choose image_editor` or `wp cache delete wp_image_editor_choose image_editor` to review or remove the values, which is more difficult with a dynamic name.

Reply:
That sounds good to me, makes sense. Let's go with this approach then.